web3/method.py

```diff
@@ -150,3 +151,16 @@ def _pipe_and_accumulate(val, fns):
     for fn in fns:
         val = fn(val)
         yield val
+
+
+class DeprecatedMethod():
+    def __init__(self, method, old_name, new_name):
+        self.method = method
+        self.old_name = old_name
+        self.new_name = new_name
+
+    def __get__(self, obj=None, obj_type=None):
+        warnings.warn(
+            "%s is deprecated in favor of %s" % (self.old_name, self.new_name),
```

Comment:
We prefer `.format(...)` style string formatting over `%` (and try to stick with a single style per codebase).

Reply:
Actually, since web3 is py3.6+ as of v5, we can now move to f-strings as the standard! :tada:

f-strings are way better than either `%` or `format()`.